File: week6/lab_java/bank/Bank.java
package bank;

import java.util.Scanner;

class Bank {
    private Customer customer;
    private Scanner in = new Scanner(System.in);
    
    public static void main(String[] args) {
        (new Bank()).menu();
    }

    public Bank() {
        this.customer = new Customer("John Smith");
    }

    private double readAmount(String action) {
        System.out.print("Amount to " + action + " $");
        return in.nextDouble();
    }

    private void deposit() {
        double amount = readAmount("deposit");
        this.customer.deposit(amount);
        System.out.printf("Amount %.2f deposited%n", amount);
    }

    private void withdraw() {
        double amount = readAmount("withdraw");
        if (customer.isSufficient(amount)) {
            customer.withdraw(amount);
            System.out.printf("Amount %.2f withdrawn%n", amount);
        } else {
            System.out.println("Insufficient funds!");
        }
    }

    private void transfer() {
        double amount = readAmount("transfer");
        if (customer.isSufficient(amount)) {
            customer.transfer(amount);
            System.out.printf("Amount %.2f transferred %n", amount);
        } else {
            System.out.println("Insufficient funds!");
        }
    }

    private void show() {
        customer.show();
    }

    private char readChoice() {
        System.out.print("Start Banking (d/w/t/s/x): ");
        return in.next().charAt(0);
    }
    
    private void help() {
        System.out.println("d - deposit");
        System.out.println("w - withdraw");
        System.out.println("t - transfer");
        System.out.println("s - show");
        System.out.println("x - exit");
    }
    
    private void menu() {
        char c;
        while ((c=readChoice()) != 'x') {
            switch (c) {
                case 'd':
                    deposit();
                    break;
                case 'w':
                    withdraw();
                    break;
                case 't':
                    transfer();
                    break;
                case 's':
                    show();
                    break;
                default:
                    help();
                    break;
            }
        }
    }
}

File: week6/lab_py/bank/bank.py
from customer import Customer

class Bank:
    def __init__(self):
        self.customer = Customer("John Smith")

    def read_amount(self, action):
        print(f"Amount to {action} $",end='')
        return float(input())

    def deposit(self):
        amount = self.read_amount("deposit")
        self.customer.deposit(amount)
        print(f'Amount {amount:.2f} deposited')

    def withdraw(self):
        amount = self.read_amount("withdraw")
        if self.customer.is_sufficient(amount):
            self.customer.withdraw(amount)
            print(f'Amount {amount:.2f} withdrawn')
        else:
            print("Insufficient funds!")

    def transfer(self):
        amount = self.read_amount("transfer")
        if self.customer.is_sufficient(amount):
            self.customer.transfer(amount)
            print(f'Amount {amount:.2f} transferred')
        else:
            print("Insufficient funds!")

    def show(self):
        self.customer.show()

    def read_choice(self):
        return input("Start Banking (d/w/t/s/x): ")[0]

    def help(self):
        print("d - deposit")
        print("w - withdraw")
        print("t - transfer")
        print("s - show")
        print("x - exit")

    def main(self):
        choice = self.read_choice()

        while choice!='x':
            match choice:
                case 'd':
                    self.deposit()
                case 'w':
                    self.withdraw()
                case 't':
                    self.transfer()
                case 's':
                    self.show()
                case _:
                    self.help()

            choice = self.read_choice()

if __name__ == "__main__":
    Bank().main()